### DISCUSSION TITLE
Reduce pledge bias by allowing sector pledge to reduce to onboarding requirement

### DISCUSSION THREAD

**Discussion by anorth** [filecoin-project/FIPs#734] [Enhancements - Technical]
# Background

The sector initial pledge requirement is dominated by consensus pledge, calculated as a power-based pro-rata share of circulating supply. Fluctuations in the network power and circulating supply over time result in sectors with equivalent power, onboarded at different times, having quite different pledge amounts.

Initial pledge per unit power is fixed at the time of onboarding. It can increase if a sector is extended or upgraded when the initial pledge requirements are higher, but can never decrease: each sector’s pledge is strictly increasing. Sectors that are onboarded during a period of high circulating supply or low network power are thus locked in to that high pledge, while sectors onboarded later might enjoy a much lower initial pledge.

# Motivation for change

A relatively high sector pledge is an **incentive to expire or terminate a sector**, because the pledge (less any termination fee) is returned to the SP. Even if a provider wishes to continue participation, incentives push them to expire sectors and seal new ones rather than simply maintain their current storage. This is sometimes pejoratively termed *pledge arbitrage,* but SPs are just responding to the incentives provided by the network.

This relative incentive to terminate is **independent of a sector’s value to network participants**. Sector value is not clearly defined, but, e.g. a sector carrying client data is more valuable than CC (at least to the client), and a sector for which a client is paying a high fee might be inferred to be more valuable than one with a low fee (as revealed by the clients’ behaviour).

When a network-valuable sector (e.g. hosting a client deal) also bears a high pledge requirement, the SP faces an incentive to terminate the deal too in order to re-onboard a new sector at a lower pledge, gaining greater returns or lower borrowing costs.

Heterogeneous cost-of-power between sectors can thus distort an SP’s alignment with network outcomes. If an SP wishes to offboard sectors, it would be better for incentives to align with offboarding the sectors that are least valuable to the network. Heterogenous but locked-in pledge requirements disrupt this alignment.

# Proposal

Remove the monotonic requirement on a sector’s initial pledge, allowing it to decrease. When a sector is extended or updated, recalculate the initial pledge amount and refund to the SP if the new requirement is lower than the old requirement.

Remove the minimum sector extension duration requirement, allowing arbitrarily short extensions, including zero. This allows pledge recalculation on a sector that is already at maximum commitment. Continue to require a 180 minimum *initial* commitment duration, but allow shorter extensions.

# Discussion

The primary motivation for this proposal is to remove differential incentives for different sectors based on onboarding epoch. This will improve SPs ability to align their actions with value to the network, when that value is transmitted to them though other mechanisms (e.g. FIL+, client payments). In particular, it will remove incentives to terminate or expire sectors that were committed in times of relatively high initial pledge requirements.

### SP incentives

Another benefit is to reduce the disincentive to long sector commitments. At present, a long commitment might mean locking in a high pledge requirement, where a shorter commitment allows an SP the option to expire and re-onboard the same power at a later time for lesser pledge. Allowing a sector’s pledge amount to be reduced to the network’s current requirement at any time removes the risk of locking in high pledge. SPs can make long commitments  confident they will not be disadvantaged relative to shorter ones.

A further benefit is to remove incentives to delay onboarding when pledge requirements are falling. At present, SPs may benefit from delaying onboarding when pledge requirements are falling rapidly (the benefit depends on SP borrowing costs and rewards at the time). Allowing sector pledge to be reduced removes costs associated with committing early.

[Projections suggest](https://colab.research.google.com/drive/1qaLM2Fm27kQ07jIGBPdkdoTvo9-YAAtq?usp=sharing#scrollTo=wfSUMfMYbzn9) that sector initial pledge requirements may decrease quite significantly over the near- to medium-term. This could quickly put a very large proportion of existing sectors and power into the position of being more expensive to maintain than new onboarding, in turn threatening the rationality of maintaining client data. Expectations of this reduced pledge could also induce delayed onboarding and short commitments, reducing network stability.

### Circulating supply

Allowing refund of over-pledged sectors raises questions about the potential impact on token circulating supply. In the first degree, the consensus pledge targeting mechanism is already constructed to restore the total locked amount, given sufficient onboarding.

- When the initial pledge requirement falls, previously-committed sectors may become over-pledged
- When locked tokens are released from over-pledged sectors, the circulating supply increases
- The consensus pledge requirement for new tokens increases accordingly, to target 30% of the now-larger supply
- New onboarding locks more tokens, restoring the circulating supply.

Note that the upward response of the initial pledge requirement to the increased supply counteracts the downward movement of initial pledge requirement that triggered action in the first place. Note also that SPs are incentivised to onboard when pledge requirements fall, before they increase again due to pledge re-calculations; both individual SPs and the network as a whole end up with more committed storage for the fixed pledge.

More modelling is needed here. It’s not clear that the consensus pledge target mechanism is effective if onboarding rates are very high or very low (clearly it can’t lock tokens if there is no onboarding). This proposal may slightly increase the minimum onboarding rate required for the pledge target to remain effective (but it also improves incentives for onboarding). Experiments so far suggest that allowing pledge recalculation has no noticeable effect on circulating supply when onboarding is within some reasonable range of the replacement rate. And when onboarding is very low, the mechanism has no effect (because per-sector initial pledge trends upwards).

### Alternatives

Sector termination penalties act as some friction to pledge arbitrage, reducing the returns. But sector termination penalties also face exactly the same issue: a sector’s termination penalty amount depends on network conditions at the time of onboarding, which does not necessarily correlate with the sector’s value to network participants. Even if this were resolved with a simpler penalty calculation, termination penalties are of limited effectiveness and bring other drawbacks. 

- Termination penalties are only effective against premature termination
- Termination penalties incentivise short commitments
- Termination penalties increase risk for SPs and their capital providers

More discussion about termination penalties is at #691.

An alternative direction is to break the strict coupling between sector commitments and pledge amounts. This could remove pledge-based incentives to expire or terminate *specific* sectors. This is a promising direction, but a much more involved change for the protocol and SPs to digest. It’s worth exploring, but the theoretical possibility of a better construction shouldn’t prevent us from realising simple improvements today.

**Comment by anorth** [comment]
Here is the distribution of pledge cost at epoch 2,900,000. The current initial value was then about 0.21 FIL / 32GiB.

![Total pledge bucketed by cost-per-32GiB](https://github.com/filecoin-project/FIPs/assets/445306/57c3c93a-b5f1-4043-868e-97da92d27424)

This has "improved" quite a lot over the past months, meaning that the right-hand side tail of high pledge amounts used to be much larger. But this improvement reflects sectors having expired or been terminated since that time.

**Comment by Fatman13** [comment]
As more people are extending DCs claim, a common issue that 9x pledge is not returned when a DC sector becomes a CC sector has been observed on the network either due to not as intuitive DC extension protocol enforcement rules or misusage of commands by SPs. Was [prompted](https://github.com/filecoin-project/FIPs/discussions/313#discussioncomment-8977861) that this FIP could be the key to address this [pledge not being returned legacy issue](https://github.com/filecoin-project/FIPs/discussions/367). It would be nice to see this gets fixed in the next network upgrade.

**Reply by Fatman13** [reply]
bump for visibility.
